### PR TITLE
Docs: Add info about analyzer incompatibilities

### PR DESCRIPTION
**Pull request by novikd**
### Changelog category (leave one):

- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add documentation entry about new query analysis incompatibilities.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

<details>
    <summary>CI Settings</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step
- [ ]  Allow: Integration Tests
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Unit tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All with aarch64
- [ ]  Allow: All with ASAN
- [ ]  Allow: All with TSAN
- [ ]  Allow: All with Analyzer
- [ ]  Allow: All with Azure
- [ ]  Allow: Add your option here
---
- [ ]  Exclude: Fast test
- [ ]  Exclude: Integration Tests
- [ ]  Exclude: Stateless tests
- [ ]  Exclude: Stateful tests
- [ ]  Exclude: Performance tests
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN
- [ ]  Exclude: All with MSAN
- [ ]  Exclude: All with UBSAN
- [ ]  Exclude: All with Coverage
- [ ]  Exclude: All with Aarch64
---
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)
- [ ]  allow: batch 1 for multi-batch jobs
- [ ]  allow: batch 2
- [ ]  allow: batch 3
- [ ]  allow: batch 4, 5 and 6
</details>

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 49962e7b6e4c444bb9f2599d7a380bfdf273b231 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64643/49962e7b6e4c444bb9f2599d7a380bfdf273b231/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: docs/en/operations/analyzer.md
Comment (novikd) [inline docs/en/operations/analyzer.md:38]: This part still requires some work.

**Comment by melvynator** [comment]
Thanks for adding all these details! 

We have this doc page already: https://clickhouse.com/docs/en/guides/developer/understanding-query-execution-with-the-analyzer#analyzer

We could maybe enrich the existing one, or alternatively, we could add a link to the page you are putting together to the existing doc.

**Comment by novikd** [comment]
@melvynator I saw that documentation page and decided that it's better to keep specifics on a separate page, but I like the idea to add a link.

File: docs/en/operations/analyzer.md
Comment (vdimir) [inline docs/en/operations/analyzer.md]: Looks like example from `Columns names are changed for some queries` is fixed, at least mentioned example https://fiddle.clickhouse.com/b4b4a7b1-67e6-47f7-a3a8-a2ef3095679a in `head` returns the same result.

Comment (vdimir) [reply docs/en/operations/analyzer.md]: For first two point I added info to this branch, because it's impossible to create suggestions in PR with code inside

File: docs/en/operations/analyzer.md
Comment (vdimir) [inline docs/en/operations/analyzer.md:46]:
```suggestion
The previous query analyzer fixed this query by moving the `number > 5` filter from the `HAVING` clause to the `WHERE` clause. To fix the query, you should move all conditions that apply to non-aggregated columns to the `WHERE` section to conform to standard SQL syntax.```

**Comment by novikd** [comment]
@justindeguzman This PR is ready for review and merge. All that is left is only your review and approval.

**Comment by justindeguzman** [comment]
Looks good, made some minor changes. After this is merged, I'll modify clickhouse-docs to add it to the navigation.

**Review by justindeguzman — APPROVED** [review]

**Pull request merged**
